components/script/dom/document.rs

```diff
@@ -2413,24 +2413,23 @@ impl Document {
         // Step 2
         // > For each item target in doc’s pending scroll event targets, in the order they
         // > were added to the list, run these substeps:
-        for target in self.pending_scroll_event_targets.borrow().iter() {
+        // Step 3.
+        // > Empty doc’s pending scroll event targets.
+        // Since the scroll event callback could trigger another scroll event, we are taking all of the
+        // current scroll event to avoid borrow checking error.
+        rooted_vec!(let notify_list <- self.pending_scroll_event_targets.clone().take().into_iter());
+        for target in notify_list.iter() {
             // Step 2.1
```

Comment:
nit: please move the spec text inside the if branch to match style with the else branch